src/function-builder.ts

```diff
@@ -120,6 +121,61 @@ function assertRuntimeOptionsValid(runtimeOptions: RuntimeOptions): boolean {
       `serviceAccount must be set to 'default', a service account email, or '{serviceAccountName}@'`
     );
   }
+
+  if (runtimeOptions.labels) {
+    if (Object.keys(runtimeOptions.labels).length > MAX_NUMBER_USER_LABELS) {
+      throw new Error(
+        `A function must not have more than ${MAX_NUMBER_USER_LABELS} user-defined labels.`
+      );
+    }
+
+    const invalidLengthKeys = Object.keys(runtimeOptions.labels).filter(
+      (label) => label.length < 1 || label.length > 63
+    );
+    if (invalidLengthKeys.length > 0) {
+      throw new Error(
+        `Invalid labels: ${invalidLengthKeys.join(
+          ', '
+        )}. Label keys must be between 1 and 63 characters in length.`
+      );
+    }
+
+    const invalidLengthValues = Object.values(runtimeOptions.labels).filter(
+      (value) => value.length > 63
+    );
+    if (invalidLengthValues.length > 0) {
+      throw new Error(
+        `Invalid labels: ${invalidLengthValues.join(
+          ', '
+        )}. Label values must be less than 64 charcters.`
+      );
+    }
+
+    // Keys can contain lowercase letters, foreign characters, numbers, _ or -. They must start with a letter.
```

Comment:
nit: `They must start with a lowercase letter`

Also, I'd appreciate if we link https://cloud.google.com/resource-manager/docs/creating-managing-labels#requirements somewhere!

Reply:
Good call, I'll add that link. However, they can start with an international character as well -technically, anything in the Unicode groups 'letters lowercase' or 'letters other'